source/fundamentals/connection/connection-storms.txt

```diff
@@ -0,0 +1,145 @@
+   refusal of new connections.
+
+Avoiding Connection Storms
+--------------------------
+
+A combination of settings and methods can help to mitigate connection storms.
+These include:
+
+- **maxConnecting**: The maximum number of connections a Pool may be
+  establishing concurrently.
+
+- **socketTimeoutMS**: The maximum time the driver will wait for socket
+  reads.
+
+- **maxTime()**: The *cumulative* time limit for processing operations on
+  a MongoDB server.
+
+**maxConnecting**
++++++++++++++++++
+
+``maxConnecting`` specifies the maximum number of connections a Pool may be establishing
+concurrently. The default is 2.
+
+This is not to be confused with ``maxPoolSize``, which specifies the maximum
+number of connections that the driver maintains in the connection pool.
+
+.. example::
+
+   Consider a new application instance connecting to a single MongoDB server, using
+   the default values of **2** for ``maxConnecting`` and **100** for ``maxPoolSize``.
+   If it takes ~50 milliseconds to establish a connection, the connection pool
+   would take 2.5 seconds (50 milliseconds [latency] * 100 [``maxPoolSize``] / 2 [``maxConnecting``])
+   to fill.
+
+   In another scenario, consider an application server connecting to a 3 member
+   replica set, using the default values for ``maxConnecting`` and ``maxPoolSize``.
+   The application server creates a connection pool to each member of the replica set.
+   If it takes ~50 milliseconds to establish a connection, the connection pools
+   would still take 2.5 seconds to fill, even though a total of 300 connections will
+   be created across three connection pools.
+
+Specify ``maxConnecting`` in your connection URI or with the
+`ConnectionPoolSettings.Builder maxConnecting() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ConnectionPoolSettings.Builder.html#maxConnecting(int)>`__ method.
+
+
+**socketTimeoutMS**
++++++++++++++++++++
+
+``socketTimeoutMS`` specifies the maximum time the driver will wait for socket reads.
+The default is **0**, meaning no timeout.
+
+Setting this to a value that is too low results in **connection thrashing**,
+where the socket is closed and removed from the connection pool, and a new
+connection is initiated. For example, setting a timeout of 200 milliseconds
+and issuing a query that takes 300 milliseconds to complete would result in
+the socket being closed and a new connection being initiated.
+
+.. warning:: Do not use ``socketTimeoutMS`` to abort operations.
+
+   Some users have incorrectly used this option to try to abort operations
+   which run for too long and slow down other operations. Closing a socket
+   forces a reconnect, which slows all operations, and does not have any
+   impact on the operation that the developer is attempting to abort. MongoDB will continue
+   to process the operation, potentially causing errors if the driver attempts to
+   retry the operation.
```

Comment:
Should we mention here to use `maxTime()` instead? (even though we talk about it in the tip right below)